### PR TITLE
V2 0

### PR DESCRIPTION
**Pull request by damacus**
## Description

Update circleci orb, kitchen config, remove the server recipe

### Issues Resolved

[List any existing issues this PR resolves]

### Check List
- [ ] All tests pass. See https://github.com/sous-chefs/apache2/blob/master/TESTING.md
- [ ] New functionality includes testing.
- [ ] New functionality has been documented in the README if applicable

---
This change is [Reviewable](https://reviewable.io/reviews/sous-chefs/samba/97)

### CODE REVIEW

File: libraries/helpers.rb [hunk 1]
```diff
@@ -0,0 +1,12 @@
+class Chef
+  module Samba
+    def self.log_dir
+      case node['platform_family']
+      when 'rhel', 'fedora', 'amazon', 'suse'
+        '/var/log/samba/log.%m'
```
Comment (Xorima) [inline libraries/helpers.rb]: Why don't we just have an opinion as to how this should be?

File: resources/server.rb [hunk 2]
```diff
@@ -34,14 +16,7 @@
...
-  case node['platform_family']
-  when 'rhel', 'fedora', 'amazon', 'suse'
-    '/var/log/samba/log.%m'
-  else
-    '/var/log/samba/%m.log'
-  end
-}
+property :log_dir, String, default: lazy { Chef::Samba.log_dir }
 property :max_log_size, [String, Integer], default: '5000' # 5M
 property :options, [Hash, nil], default: nil
```
Comment (Xorima) [inline resources/server.rb]: Why not default to an empty hash?

File: templates/smb.conf.erb [hunk 3]
```diff
@@ -27,6 +27,7 @@
   dns proxy = <%= @dns_proxy %>
   max log size = <%= @max_log_size %>
   bind interfaces only = <%= @bind_interfaces_only %>
+  log level = <%= @log_level %>
```
Comment (Xorima) [inline templates/smb.conf.erb]: Do we have a default set on this somewhere?

**Comment by Xorima** [comment]
@damacus rebase and migrate your circle changes to actions please?

File: spec/unit/recipes/default_spec.rb [hunk 4]
```diff
@@ -1,13 +1,10 @@
-require 'chefspec'
-require 'chefspec/berkshelf'
+require 'spec_helper'
 
-describe 'samba::server' do
-  let(:chef_run) do
-    ChefSpec::SoloRunner.new(platform: 'ubuntu', version: '16.04').converge(described_recipe)
-  end
+describe 'Default recipe on Ubuntu 16.04' do
+  let(:runner) { ChefSpec::ServerRunner.new(platform: 'ubuntu', version: '16.04', step_into: ['samba_server']) }
```
Comment (tas50) [inline spec/unit/recipes/default_spec.rb]: Take out version here

File: libraries/helpers.rb [hunk 5]
```diff
@@ -0,0 +1,12 @@
+class Chef
+  module Samba
+    def self.log_dir
+      case node['platform_family']
```
Comment (tas50) [inline libraries/helpers.rb]: This should just be platform_family?(whatever) and else

**Comment by damacus** [comment]
If someone wants to take on this PR be my guest, however I don't really have the energy to take this one on anymore 👍

**Pull request closed without merging**